tests/zstreamtest.c

```diff
@@ -1834,6 +1834,97 @@ static int basicUnitTests(U32 seed, double compressibility, int bigTests)
     }
     DISPLAYLEVEL(3, "OK \n");
 
+    DISPLAYLEVEL(3, "test%3i : External matchfinder API: ", testNb++);
+    {
```

Comment:
Coding style : 
It seems you could write this entire unit test into its own function,
and then invoke this newly created test function from here .

I know most of the existing code base in this unit is not written using this style,
but this is probably and cleaner way to proceed, more maintainable.

Reply:
Unfortunately, this isn't possible without a larger refactor because of how the `CHECK` and `CHECK_Z` macros are defined. For example, they depend on the `_output_error` label which is defined inside `basicUnitTests()`.

Compile errors when I tried to move the test into its own function: https://gist.github.com/embg/cc39f0a37bddb57cebbace40374026ca

Reply:
Indeed, I had to refactor these macros in `fuzzer.c` in order to write new unit tests as independent functions.
It's not a difficult effort, but I agree that it's a _different_ effort, and is better kept as a separate topic.
Maybe a follow up.